src/common/types/store-data/unified-data-interface.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// this is similar to `TestEngine` interface from axe-core
+export interface ScanEngine {
+    name: string;
+    version: string;
+}
+
+export interface OSData {
+    name: string;
+    osVersion?: string;
+    userAgent?: string;
+}
+
+export interface ViewPortData {
+    platformHeight?: number;
+    platformWidth?: number;
+}
+
+export interface PlatformData {
+    osInfo: OSData;
+    viewPortInfo: ViewPortData;
+}
+
+export interface ToolData {
+    scanEngine: ScanEngine;
+}
+
+export interface ScanResults {
+    results: UnifiedResultInstance[];
+    platformInfo?: PlatformData;
+    toolInfo?: ToolData;
```

Comment:
Is it ever possible not to know the platform or the tool?

Reply:
They are optional because we don't get this data from all sources as of now. For example, in [axe-windows/docs/AutomationReference](https://github.com/microsoft/axe-windows/blob/master/docs/AutomationReference.md), we currently don't receive tool or platform information. However, we expect to get this information in the future.